### ISSUE TITLE
Using within_rectangle is slower than 2 skiplist indexes

### ISSUE THREAD

**Issue by matcho** [arangodb/arangodb#5936]
## My Environment

* __ArangoDB Version__:        3.3.11
* __Storage Engine__:            RocksDB
* __Deployment Mode__:       Single Server
* __Infrastructure__:               own
* __Operating System__:        Debian 9
* __Used Package__:              Debian or Ubuntu .deb

## Component, Query & Data

__Affected feature__:
AQL query (whatever the driver / interface)

__AQL query 1__:
```
FOR p IN WITHIN_RECTANGLE(geotest, 10.1, 0.9, 75.2, 143) RETURN p.note
```

__AQL query 2__:
```
FOR p IN geotest
    FILTER p.lat < 75.2 
    FILTER p.lat > 10.1
    FILTER p.lon < 143
    FILTER p.lon > 0.9
    RETURN p.note 
```

__Dataset__:
100 000 randomly generated geolocated data, with one geo2 index and 2 skiplist indexes on `lat` and `lon`
```
FOR i IN RANGE(1, 100000)
    INSERT {
        lat: RAND() * 90,
        lon: RAND() * 180,
        note: CONCAT("point number ", i)
    } IN geotest
```
Indexes :
```
ID 		Type 		Unique 	Sparse 	Deduplicate 	Selectivity Est. 	Fields
0		primary		true	false	n/a		100.00%			_key	
10744633355	geo2		false	true	n/a		n/a			lat, lon
10744633389	skiplist	false	false	false		99.99%			lat
10744633432	skiplist	false	false	false		100.00%			lon
```

## Steps to reproduce

1. Generate dataset and create indexes
2. Run query 1 (within_rectangle)
3. Run query 2 (inequalities on coordinates)

__Problem__:
Query 1 is significantly slower than query 2 (3.169 s / 708.291 ms); in my case they both return the same 57 496 documents.

__Expected result__:
A dedicated geographical query operator like `WITHIN_RECTANGLE` is expected to be faster than the "trivial" solution; while this is not a big deal this is quite surprising.
With small datasets the speed is approximately the same, and the gap gets wider when dataset increases.

**Comment by Simran-B** [comment]
In general, there are no collection-level locks if you use the RocksDB storage engine, but in case of the geo index there are collection-level write locks (see [Known Limitations](https://docs.arangodb.com/3.3/Manual/ReleaseNotes/KnownIssues32.html)). This is required for the current geo index implementation to work with RocksDB, which was written when there was only MMFiles storage engine. It will be resolved with 3.4, which will come with a new geo index implementation and also additional geo features. You can already find that work in devel branch.

**Comment by vinaypyati** [comment]
Hi @matcho,

Hope that explains your questions. I am marking this issue as solved. Please feel free to reopen this issue if you have any further queries.

Thanks!
